lametro/models.py

```diff
@@ -191,29 +191,58 @@ def upcoming_board_meeting(cls):
 
     @classmethod
     def current_meeting(cls):
-        # Create the boundaries for discovering events (in progess) within the timeframe stipulated 
-        # by Metro.
-        # Then, filter the events according to these boundaries.
-        six_minutes_from_now = datetime.now(app_timezone) + timedelta(minutes=6)
-        three_hours_ago = datetime.now(app_timezone) - timedelta(hours=3)
+        # QA Testing....
+        # fakenow = datetime(2018,1,18,10,10)
+        # six_minutes_from_now = fakenow + timedelta(minutes=6)
+        # three_hours_ago = fakenow - timedelta(hours=6)
+
+        # We look for everything in the last 6 hours, because the maximum recorded meeting is 5.38 hours (according to the spreadsheet provided by Metro in issue #251).
+        six_minutes_from_now = datetime.now(app_timezone) + timedelta(minutes=5)
+        three_hours_ago = datetime.now(app_timezone) - timedelta(hours=6)
+        found_events = cls.objects.filter(start_time__lte=six_minutes_from_now)\
+                  .filter(start_time__gte=three_hours_ago)\
+                  .exclude(status='cancelled')\
+                  .order_by('start_time')
+
 
+        '''
+        Create the boundaries for discovering events (in progess) within the timeframe stipulated 
+        by Metro. A meeting displays as current if:
+        (1) "now" is six minutes or less before the designated start time;
+        (2) the previous meeting has ended (determined by looking for "In progress" on Legistar);
+
+        This method returns a list (with zero or more elements).
+        '''
+        # six_minutes_from_now = datetime.now(app_timezone) + timedelta(minutes=6)
+        # three_hours_ago = datetime.now(app_timezone) - timedelta(hours=6)
         found_events = cls.objects.filter(start_time__lt=six_minutes_from_now)\
                   .filter(start_time__gt=three_hours_ago)\
                   .exclude(status='cancelled')\
                   .order_by('start_time')
 
+        #  Solution 1: calculate the current meeting, given particular parameters, and use Legistar to determine when a meeting has ended.
         if found_events:
-            if len(found_events) > 1: 
-                return calculate_current_meetings(found_events, six_minutes_from_now)   
-            else:
-                # A board meeting lasts 3 hours, and a committee event lasts 1 hour.
-                # Change the value of "_hour_ago" for committee meetings to 1 hour.
-                if found_events.filter(name__icontains='Board Meeting'):
-                    return found_events.first()
-                else:
-                    one_hour_ago = six_minutes_from_now - timedelta(hours=1)
-                        
-                    return found_events.filter(start_time__gt=one_hour_ago).first()
+            return calculate_current_meetings(found_events, six_minutes_from_now)
+
+        # Solution 2: rely entirely on Legistar.
```

Comment:
Did you mean to leave this in here?

Reply:
Yes, but I just moved it to a comment in the original issue.